components/dashboard/src/data/configurations/configuration-queries.ts

```diff
@@ -4,33 +4,42 @@
  * See License.AGPL.txt in the project root for license information.
  */
 
-import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { useInfiniteQuery, useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
 import { useCurrentOrg } from "../organizations/orgs-query";
 import { configurationClient } from "../../service/public-api";
 import type { Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import { useStateWithDebounce } from "../../hooks/use-state-with-debounce";
+import { useEffect } from "react";
 
 const BASE_KEY = "configurations";
 
 type ListConfigurationsArgs = {
+    pageSize?: number;
     searchTerm?: string;
-    page: number;
-    pageSize: number;
 };
 
-export const useListConfigurations = ({ searchTerm = "", page, pageSize }: ListConfigurationsArgs) => {
+export const useListConfigurations = ({ searchTerm = "", pageSize }: ListConfigurationsArgs) => {
     const { data: org } = useCurrentOrg();
 
-    return useQuery(
-        getListConfigurationsQueryKey(org?.id || "", { searchTerm, page, pageSize }),
-        async () => {
+    // Debounce searchTerm for query
```

Comment:
Chose to pull debouncing the search term into here and let the UI component just pass the value in. Felt like it kept the UI component a bit simpler.